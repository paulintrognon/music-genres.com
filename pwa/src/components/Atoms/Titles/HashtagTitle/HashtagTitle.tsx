import classnames from 'classnames'
import styles from './HashtagTitle.module.scss'

type Props = React.ComponentProps<'span'>
const HashtagTitle: React.FC<Props> = ({ className, children }) => {
  return <span className={classnames(styles.container, className)}>#&nbsp;{children}</span>
}
export default HashtagTitle
