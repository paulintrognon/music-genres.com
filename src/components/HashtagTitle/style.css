.hashtag-title {
  font-family: 'Montserrat Subrayada';
  margin-bottom: 10px;
  text-transform: uppercase;
}
