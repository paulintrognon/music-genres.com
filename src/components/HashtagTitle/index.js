import React from 'react';

import './style.css';

export default function HashtagTitle(props) {
  let className = 'hashtag-title ';

  if (props.className) {
    className = className + props.className;
  }

  if (props.clickHandler) {
    className = className + ' clickable';
  }

  return (
    <h2 className={className} onClick={props.clickHandler} title={props.title}>
      # {props.children}
    </h2>
  );
}
