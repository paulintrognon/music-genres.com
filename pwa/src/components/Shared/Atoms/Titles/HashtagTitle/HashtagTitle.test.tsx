import HashtagTitle from './HashtagTitle'
import { shallow } from 'enzyme'

it('should render the HashtagTitle', () => {
  const wrapper = shallow(<HashtagTitle className="testClass">Super Title</HashtagTitle>)
  expect(wrapper).toMatchSnapshot()
})
