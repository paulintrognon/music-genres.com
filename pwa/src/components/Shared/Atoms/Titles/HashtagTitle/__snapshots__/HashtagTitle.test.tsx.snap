// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the HashtagTitle 1`] = `
<span
  className="container testClass"
>
  #Â 
  Super Title
</span>
`;
